### ISSUE TITLE
libffi.so.8 not found

### ISSUE THREAD

**Issue by rgathmann** [ffi/ffi#947]
I am working with Ruby 2.7.5 on Ubuntu 20.04. I currently have the libffi-dev package installed, which I believe provides libffi.so.7. However when running my rails app after installing sassc-rails which depends on ffi. I get the following error:

```libffi.so.8: cannot open shared object file: No such file or directory - /home/rgathmann/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/ffi-1.15.5/lib/ffi_c.so```

I have also run the following command which shows that while the ffi_c.so was installed correctly. The dependency is indeed missing.

```
ldd /home/username/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/ffi-1.15.5/lib/ffi_c.so
        linux-vdso.so.1 (0x00007ffe75fc7000)
        libruby.so.2.7 => /home/rgathmann/.rbenv/versions/2.7.5/lib/libruby.so.2.7 (0x00007f44a2386000)
        libffi.so.8 => not found
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f44a218a000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f44a2167000)
        libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f44a214b000)
        librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f44a213f000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f44a2139000)
        libcrypt.so.1 => /lib/x86_64-linux-gnu/libcrypt.so.1 (0x00007f44a20fe000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f44a1faf000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f44a2708000)
```

My question is, can I force FFI to use the system provided libffi.so.7 instead of libffi.so.8?

**Comment by rgathmann** [comment]
After more searching, I found an answer and I understand what seems to be going on now. Since Ubuntu 20.04 is so common, I will provide my solution to hopefully someone else out in the future.  

The FFI version included with Ubuntu 20.04 in too old. So, the key is to uninstall the gem and then reinstall it with the correct flag to force it to ignore the system ffi library.

First uninstall the gem:
`gem uninstall ffi`

Then reinstall it with disable flag:
`gem install ffi -- --disable-system-libffi`

Now the ffi gem should use its own libffi library instead of the system library.

**Comment by larskanis** [comment]
Thanks for sharing your answer!
